nixos/modules/misc/mandoc.nix

```
@@ -36,28 +38,166 @@ in {
           configuration options used for the package.
         '';
       };
+
+      settings = lib.mkOption {
+        description = lib.mdDoc "Configuration for {manpage}`man.conf(5)`";
+        default = { };
+        type = lib.types.submodule {
+          options = {
+            manpath = lib.mkOption {
+              type = with lib.types; listOf str;
+              default = [ ];
+              example = lib.literalExpression "[ \"/run/current-system/sw/share/man\" ]";
+              description = lib.mdDoc ''
+                Override the default search path for {manpage}`man(1)`,
+                {manpage}`apropos(1)`, and {manpage}`makewhatis(8)`. It can be
+                used multiple times to specify multiple paths, with the order
+                determining the manual page search order.
+                This is not recommended in favor of
+                {option}`documentation.man.mandoc.manPath`, but if it's needed to
+                specify the manpath in this way, put
+                {option}`documentation.man.mandoc.manPath` to an empty list (`[]`).
+              '';
+            };
+            fragment = lib.mkEnableOption (lib.mdDoc ''
+              Omit the <!DOCTYPE> declaration and the <html>, <head>, and <body>
+              elements and only emit the subtree below the <body> element in HTML
+              output of {manpage}`mandoc(1)`. The style argument will be ignored.
+              This is useful when embedding manual content within existing documents.
+            '');
```

I think the documentation would look awkward with `mkEnableOption`, so we should override it:

```suggestion
            output.fragment = lib.mkEnableOption null // {
              description = lib.mdDoc ''
                Whether to omit the <!DOCTYPE> declaration and the <html>, <head>, and <body>
                elements and only emit the subtree below the <body> element in HTML
                output of {manpage}`mandoc(1)`. The style argument will be ignored.
                This is useful when embedding manual content within existing documents.
              ''
            };
```